sw/device/silicon_creator/lib/drivers/spi_device.c

```diff
@@ -501,10 +502,13 @@ void spi_device_init(void) {
   reg = bitfield_field32_write(reg, SPI_DEVICE_JEDEC_CC_NUM_CC_FIELD,
                                kSpiDeviceJedecContCodeCount);
   abs_mmio_write32(kBase + SPI_DEVICE_JEDEC_CC_REG_OFFSET, reg);
-  // TODO(#11605): Use the HW revision register when available.
-  reg = bitfield_field32_write(0, SPI_DEVICE_DEV_ID_CHIP_REV_FIELD, 0);
+  lifecycle_hw_rev_t hw_rev;
+  lifecycle_hw_rev_get(&hw_rev);
+  reg = bitfield_field32_write(0, SPI_DEVICE_DEV_ID_CHIP_REV_FIELD,
+                               hw_rev.chip_rev);
   reg = bitfield_bit32_write(reg, SPI_DEVICE_DEV_ID_ROM_BOOTSTRAP_BIT, true);
-  reg = bitfield_field32_write(reg, SPI_DEVICE_DEV_ID_CHIP_GEN_FIELD, 0);
+  reg = bitfield_field32_write(reg, SPI_DEVICE_DEV_ID_CHIP_GEN_FIELD,
+                               hw_rev.chip_gen);
```

Comment:
The hw_rev components are 16 bits, whereas the SPI_DEVICE_CHIP_* fields are just a few bits (3 or 4, IIRC).

We have an unstated assumption here that the chip_{rev,gen} will fit in those fields (and they _probably_ always will).  Leave a comment with this assumption.